packages/api/src/middlewares/withTelemetry.ts

```diff
@@ -1,43 +1,98 @@
-import type { api } from "@blobscan/open-telemetry";
+import { api } from "@blobscan/open-telemetry";
 
-import { tracer } from "../instrumentation";
+import { meter, tracer } from "../instrumentation";
 import { t } from "../trpc-client";
 
+type APICounter = {
+  scope?: string;
+};
+
+type RequestsTotalCounter = APICounter & {
+  endpoint: string;
+  status_code: string;
+  method: string;
+};
+
+const apiRequestsTotalCounter = meter.createCounter<RequestsTotalCounter>(
+  "blobscan_api_requests_total",
+  {
+    valueType: api.ValueType.INT,
+    description:
+      "Number of requests made to Blobscan API by endpoint, status_code and method",
+  }
+);
+
+const apiRequestsDurationMsHistogram = meter.createHistogram<APICounter>(
```

Comment:
It should be `meter.createHistogram<RequestsTotalCounter>` too if I'm not wrong

Reply:
Yep